client/src/utils/auth.js

```diff
@@ -0,0 +1,65 @@
+import 'whatwg-fetch';
+import jwtDecode from 'jwt-decode';
+
+const LOCAL_STORE_STRING = 'userJWT';
+const ADMIN_ROUTES = [
+  '/performances',
+  '/performances/new',
+  '/results',
+  '/users',
+  '/users/search'
+];
+const SL_ROUTES = [
+  '/challenges/evaluate'
+];
+const url = process.env.NODE_ENV === 'development'
+  ? `http://localhost:${process.env.SERVER_PORT}`
+  : '';
+
+
+const authenticate = (nameNumber, password) =>
+  fetch(`${url}/api/token`, {
+    headers: {
+      Accept: 'application/json, text/html', // eslint-disable-line quote-props
+      'Content-Type': 'application/json'
+    },
+    method: 'post',
+    body: JSON.stringify({ nameNumber, password })
+  })
+  .then((response) => {
+    if (response.status >= 300) throw new Error(response.status);
+    return response.json();
+  })
+  .then(({ token }) => {
+    localStorage.userJWT = token;
+  });
+
+const getToken = () => localStorage[LOCAL_STORE_STRING];
+
+const getUser = () => localStorage[LOCAL_STORE_STRING] && jwtDecode(localStorage[LOCAL_STORE_STRING]);
+
+const isAuthenticated = () => {
+  const user = getUser();
+
+  if (!user) {
+    return false;
+  } else if (new Date(user.revokeTokenDate).getTime() > user.iat || user.expires < new Date().getTime()) {
+    logout();
+    return false;
+  }
+  return true;
+};
+
+const logout = () => {
+  localStorage.removeItem(LOCAL_STORE_STRING);
+};
+
+const refreshToken = (token) => {
+  localStorage.userJWT = token;
+};
+
+const userCanAccess = (pattern) =>
```

Comment:
The route information should be in `routes.js` this function can still exist, but should probably ask the routes file about permisisons